addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/scanrules/ActiveScriptScanRule.java

```diff
@@ -0,0 +1,247 @@
+        try {
+            if (!script.isEnabled()) {
+                // The script was disabled while the scan was running (intentionally or otherwise).
+                getParent()
+                        .pluginSkipped(
+                                this,
+                                Constant.messages.getString(
+                                        "scripts.scanRules.ascan.disabledSkipReason"));
+                return;
+            }
+            ActiveScript s = cachedScriptInterfaces.getInterface(script, ActiveScript.class);
+            if (s != null) {
+                s.scan(this, msg, param, value);
+            }
+        } catch (Exception e) {
+            getExtScript().handleScriptException(script, e);
+        }
+    }
+
+    @Override
+    public void cloneInto(Plugin other) {
+        if (!(other instanceof ActiveScriptScanRule)) {
+            throw new IllegalArgumentException(
+                    "Expected a ActiveScriptScanRule, but got " + other.getClass().getName());
+        }
+        var otherRule = (ActiveScriptScanRule) other;
+        otherRule.script = script;
+        otherRule.metadata = metadata;
+        otherRule.setConfig(getConfig());
+    }
+
+    final ScriptWrapper getScript() {
+        return script;
+    }
+
+    void setMetadata(ScanRuleMetadata metadata) {
+        this.metadata = metadata;
+    }
+
+    @Override
+    public AlertBuilder newAlert() {
+        return super.newAlert()
+                .setConfidence(metadata.getConfidence().getValue())
+                .setOtherInfo(metadata.getOtherInfo());
+    }
+
+    @Override
+    public int getId() {
+        return metadata.getId();
+    }
+
+    @Override
+    public String getName() {
+        return metadata.getName();
+    }
+
+    @Override
+    public String getDescription() {
+        return metadata.getDescription();
+    }
+
+    @Override
+    public int getCategory() {
+        return metadata.getCategory().getValue();
+    }
+
+    @Override
+    public String getSolution() {
+        return metadata.getSolution();
+    }
+
+    @Override
+    public String getReference() {
+        return String.join("\n", metadata.getReferences());
+    }
+
+    @Override
+    public int getRisk() {
+        return metadata.getRisk().getValue();
+    }
+
+    @Override
+    public int getCweId() {
+        return metadata.getCweId();
+    }
+
+    @Override
+    public int getWascId() {
+        return metadata.getWascId();
+    }
+
+    @Override
+    public Map<String, String> getAlertTags() {
+        return metadata.getAlertTags();
+    }
+
+    @Override
+    public AddOn.Status getStatus() {
+        return metadata.getStatus();
+    }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return List.of(newAlert().build());
+    }
+
+    @Override
+    public void setEnabled(boolean enabled) {
+        super.setEnabled(enabled); // TODO: Is this needed?
```

Comment:
Yes, it does more than just setting the enabled state.

Reply:
Right, but the "enabled" property added to the config file is irrelevant for script scan rules, since that uses the enabled state of the script instead.

Reply:
Agreed but the base method also changes the threshold to be in sync with the enabled state.